ts/components/__tests__/TosWebviewComponent.test.tsx

```diff
@@ -0,0 +1,169 @@
+import { fireEvent, render } from "@testing-library/react-native";
+import * as React from "react";
+import WebView from "react-native-webview";
+import renderer from "react-test-renderer";
+import * as O from "fp-ts/lib/Option";
+import {
+  WebViewErrorEvent,
+  WebViewMessageEvent,
+  WebViewNavigationEvent
+} from "react-native-webview/lib/WebViewTypes";
+import TosWebviewComponent from "../TosWebviewComponent";
+
+beforeAll(() => {
+  jest.resetAllMocks();
+});
+
+afterAll(() => {
+  jest.resetAllMocks();
+});
+
+describe("TosWebviewComponent", () => {
+  describe("The snapshot for the TosWebviewComponent", () => {
+    it("Renders correctlye", () => {
+      const tree = renderer
+        .create(
+          <TosWebviewComponent
+            shouldFooterRender={true}
+            webViewSource={{ html: "<html><head></head><body></body></html>" }}
+            handleLoadEnd={() => undefined}
+            handleError={() => undefined}
+          />
+        )
+        .toJSON();
+      expect(tree).toMatchSnapshot();
+    });
+  });
+  describe("When rendering with the shouldFooterRender set to false", () => {
+    it("The footer should not render", () => {
+      const renderAPI = commonSetup({ shouldRenderFooter: false });
+
+      // The footer should be rendered
+      const footerWithButtonsViewRTI =
+        renderAPI.queryByTestId("FooterWithButtons");
+      expect(footerWithButtonsViewRTI).toBeFalsy();
+    });
+  });
+  describe("When rendering with the footer displayed", () => {
+    it("Clicking the left button should trigger 'onExit' prop handler", () => {
+      const leftButtonHandlerMock = jest.fn();
+      const renderAPI = commonSetup({ onLeftButton: leftButtonHandlerMock });
+
+      // Find the left button and press it
+      const footerWithButtonsViewRTI = renderAPI.getByTestId(
+        "toSWebViewContainerFooterLeftButton"
+      );
+      fireEvent.press(footerWithButtonsViewRTI);
+
+      // The left button handler should have been invoked
+      expect(leftButtonHandlerMock).toHaveBeenCalledTimes(1);
+    });
+  });
+  describe("When rendering with the footer displayed", () => {
```

Comment:
Same description. What do you think about move `it` under same describe?

Reply:
Yep, nice catch!